### PR TITLE
Add certificate creation and Manifest.MoM signing

### PR DESCRIPTION
**Pull request by tmarcu**
Signed-off-by: Tudor Marcu <tudor.marcu@intel.com>

### CODE REVIEW

File: build-chroots.sh [hunk 1]
```diff
@@ -106,8 +106,16 @@ else
...
 
-# insert the certificate needed for signing verification
+# Create the certificate needed for signing verification if it does not exist, and then
+# insert it into the chroot
 if [ ! -z $CERT ]; then
+    if [ ! -f $CERT ]; then
+        # This generates the private key and self signed certificate
+        openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 \
+                -keyout private.pem -out $CERT \
+                -subj "/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=www.example.com/DN=MixerCert" \
```
Comment (phmccarty) [inline build-chroots.sh]: Should all of these -subj fields be configurable? Or is that added later?

File: create-update.sh [hunk 2]
```diff
@@ -102,6 +102,13 @@ if [  "$KEEP_CHROOTS" -eq 0 ]; then
 		done
 fi
 
+# step 1.5: sign the Manifest.MoM that was just created
+if [ -f "ClearLinuxRoot.pem" ]; then
+		sudo openssl smime -sign -binary -in update/www/$MIXVER/Manifest.MoM \
```
Comment (phmccarty) [inline create-update.sh]: The indentation level looks off here.

File: build-chroots.sh [hunk 3]
```diff
@@ -106,8 +106,16 @@ else
     sudo -E sh -c "LD_PRELOAD=/usr/lib64/nosync/nosync.so $BUILDERSCRIPT -m $BUILDVER $CLRVER"
 fi
 
-# insert the certificate needed for signing verification
+# Create the certificate needed for signing verification if it does not exist, and then
+# insert it into the chroot
 if [ ! -z $CERT ]; then
+    if [ ! -f $CERT ]; then
+        # This generates the private key and self signed certificate
+        openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 \
```
Comment (phmccarty) [inline build-chroots.sh]: Some users may want a shorter/longer cert lifetime than 1 year.

File: create-update.sh (hunk 2 above)
Comment (tmarcu) [inline create-update.sh]: Thanks for the comments, and yes the fields (date, subj, etc metadata) will be configurable soon, I just wanted this to be a first pass at enabling the feature so users who just want it to work - have it working.

Comment (phmccarty) [inline create-update.sh]: Okay, sounds good.

**Comment by tmarcu** [comment]
Merged 139b693

**Pull request closed without merging**